record/record.go

```diff
@@ -52,23 +52,29 @@ func RemapRecord(inputRecord FluentBitRecord, inputTimestamp interface{}, plugin
 // parseRecord transforms a log record emitted by FluentBit into a LogRecord
 // domain type: a map of string keys and arbitrary (int, string, etc.) values.
 // No value modification is performed by this method (except casting).
-func parseRecord(inputRecord FluentBitRecord) (outputRecord LogRecord) {
-	outputRecord = make(map[string]interface{})
-	for k, v := range inputRecord {
-		switch value := v.(type) {
-		case []byte:
-			outputRecord[k.(string)] = string(value)
-			break
-		case string:
-			outputRecord[k.(string)] = value
-			break
-		case map[interface{}]interface{}:
-			outputRecord[k.(string)] = parseRecord(value)
-		default:
-			outputRecord[k.(string)] = value
+func parseRecord(inputRecord map[interface{}]interface{}) map[string]interface{} {
```

Comment:
I think that we can keep return the type `LogRecord` no?

Reply:
Discussed via slack